src/main/java/org/folio/orders/utils/HelperUtils.java

```diff
@@ -1097,4 +1097,88 @@ public static PoLine convertToPoLine(CompositePoLine compPoLine) {
+        try {
+          if (isFieldChanged(compPO, compPOFromStorage, field)) {
+            fields.add(field);
+          }
+        } catch (IllegalAccessException e) {
+          throw new CompletionException(e);
+        }
+      }
+
+      verifyProtectedFieldsChanged(fields);
+    }
+
+    return completedFuture(compPOFromStorage);
+
+  }
+
+  public static Set<String> findChangedPoLineProtectedFields(CompositePoLine compPOLine, CompositePoLine compPOLineFromStorage) {
+    Set<String> fields = new HashSet<>();
+    for (String field : POLineProtectedFields.getFieldNames()) {
+      try {
+        if (field.contains(".")) {
+          String[] subObject = field.split(FIELD_SEPARATOR);
+          if (isSubObjectAdded(compPOLine, compPOLineFromStorage, subObject[0])
+              || isSubObjectDeleted(compPOLine, compPOLineFromStorage, subObject[0])
+              || isSubObjectFieldChanged(compPOLine, compPOLineFromStorage, subObject)) {
+            fields.add(field);
+          }
+        } else if (isFieldChanged(compPOLine, compPOLineFromStorage, field)) {
+          fields.add(field);
+        }
+      } catch (IllegalAccessException e) {
+        throw new CompletionException(e);
+      }
+    }
+    return fields;
+  }
+
+  public static void verifyProtectedFieldsChanged(Set<String> fields) {
+    if (CollectionUtils.isNotEmpty(fields)) {
+      Error error = PROHIBITED_FIELD_CHANGING.toError()
+        .withAdditionalProperty(PROTECTED_AND_MODIFIED_FIELDS, fields);
+      throw new HttpException(400, error);
+    }
+  }
+
+  private static boolean isSubObjectFieldChanged(CompositePoLine compPOLine, CompositePoLine compPOLineFromStorage,
+      String[] subObject) throws IllegalAccessException {
+    return !isFieldNull(compPOLine, subObject[0]) && !isFieldNull(compPOLineFromStorage, subObject[0])
+        && isFieldChanged(readDeclaredField(compPOLine, subObject[0], true),
+            readDeclaredField(compPOLineFromStorage, subObject[0], true), subObject[1]);
+  }
+
+  private static boolean isSubObjectDeleted(CompositePoLine compPOLine, CompositePoLine compPOLineFromStorage, String splits)
+      throws IllegalAccessException {
+    return isFieldNull(compPOLine, splits) && !isFieldNull(compPOLineFromStorage, splits);
+  }
+
+  private static boolean isSubObjectAdded(CompositePoLine compPOLine, CompositePoLine compPOLineFromStorage, String splits)
+      throws IllegalAccessException {
+    return !isFieldNull(compPOLine, splits) && isFieldNull(compPOLineFromStorage, splits);
+  }
+
+  private static boolean isFieldNull(CompositePoLine compPoLine, String splits) throws IllegalAccessException {
+    return null == readDeclaredField(compPoLine, splits, true);
+  }
+
+  private static boolean isFieldChanged(Object newObject, Object existedObject, String field) throws IllegalAccessException {
+    return !EqualsBuilder.reflectionEquals(FieldUtils.readDeclaredField(newObject, field, true),
```

Comment:
`IllegalArgumentException` can be thrown if `field` is not present in the `newObject` or `existedObject`.

Reply:
I remember clearly that I caught such an exception when I was working on API tests for mod-invoice. I'm not sure how this is possible, because when RMB deserialize json into a POJO it initializes missing fields with null or empty list. 
You can ignore the previous comment because I couldn't reproduce it.